### DISCUSSION TITLE
What version of `ruff` do I use?

### DISCUSSION THREAD

**Discussion by phi-friday** [astral-sh/rye#827] [Q&A] [answered]
I know that `rye` uses `ruff` when `lint` and `fmt` are used.
In this case, what version of `ruff` is used by `rye`?
And if `pyproject.toml` has `ruff` added to it, does `rye` use `ruff` in `.venv`?

**Comment by bluss** [comment, marked as answer]
You could peek into rye's self environment and check what it has there right now like this:

```shell
./.rye/self/bin/ruff --version
```
```
ruff 0.2.2
```

The next release of rye will bump ruff to 0.3.0 


And rye always calls the ruff from its 'self' virtualenv.

Not so easy to design I think but I would also expect that more flexibility is needed in the future w.r.t which tool versions to use and where they are installed.

**Reply by phi-friday** [reply]
To summarize my understanding,

1. `ruff` is built-in.
2. if `ruff` is installed on `.venv`, use the installed `ruff`.
3. behavior 2 is the intended behavior.

If this is true, I can use `rye` without any worries.
Am I understanding this correctly?

**Reply by phi-friday** [reply]
ah..
When you say `self` venv, I assume you mean the built-in `ruff` and not the one I installed?

**Reply by phi-friday** [reply]
this one takes a little thought.
thanks for the answer.
have a good day.
